include/emp/base/optional_throw.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+ *  This file is part of Empirical, https://github.com/devosoft/Empirical
+ *  Copyright (C) Michigan State University, MIT Software license; see doc/LICENSE.md
+ *  date: 2024
+*/
+/**
+ *  @file
+ *  @brief TODO.
+ *
+ */
+
+#ifndef EMP_BASE_OPTIONAL_THROW_HPP_INCLUDE
+#define EMP_BASE_OPTIONAL_THROW_HPP_INCLUDE
+
+#include "assert.hpp"
+
+/// NDEBUG should trigger its EMP equivalent.
+#ifdef NDEBUG
+#define EMP_NDEBUG
+#endif
+
+
+#if defined( EMP_OPTIONAL_THROW_ON )
+
+  // #if defined (_MSC_VER )
+
+  #define emp_optional_throw(TEST, MESSAGE)                                     \
+    do {                                                                  \
+      if (!(TEST)) {                                                      \
+        emp::assert_throw_opt(__FILE__, __LINE__, #TEST, MESSAGE, 0);                \
+      }                                                                   \
+    } while(0)
+
+  // #define emp_optional_throw_impl(TEST, MESSAGE) emp_optional_throw_mscv_impl(TEST, MESSAGE)
```

Comment:
I could, but this is code that I'm hoping to be able to go back to one day. I had to simplify to a worse version of optional_throw for windows compatibility. This comment out code is the path towards handling that better. I don't have time to fix it right now but it would be great if someone did one day